doc/source/how-to/contributing.rst

```diff
@@ -0,0 +1,249 @@
+
+2. Update `pip` by running:
+
+   .. code-block:: bash
+
+       python -m pip install --upgrade pip
+
+3. Install the library using:
+
+   .. code-block:: bash
+
+       python -m pip install --editable .
+
+
+Creating a New Branch
+---------------------
+It is likely that the default branch name is ``main`` or ``master``. This is the
+development branch, considering the :ref:`Branching Model` followed in PyAnsys
+projects. 
+
+New contributions need to be implemented in a different branch. Then, these
+changes are merged into ``main`` by :ref:`Creating a Pull Request`. To create a
+new branch run:
+
+.. code-block:: bash
+
+    git checkout -b <new branch name>
+
+Branch Naming Conventions
+~~~~~~~~~~~~~~~~~~~~~~~~~
+To streamline development, we have the following requirements for naming
+branches. These requirements help core developers know what kind of
+changes any given branch is introducing before looking at the code.
+
+-  ``fix/``: any bug fixes, patches, or experimental changes that are
+   minor
+-  ``feat/``: any changes that introduce a new feature or significant
+   addition
+-  ``junk/``: for any experimental changes that can be deleted if gone
+   stale
+-  ``maint/``: for general maintenance of the repository or CI routines
+-  ``doc/``: for any changes only pertaining to documentation
+-  ``no-ci/``: for low-impact activity that should not trigger the CI
+   routines
+-  ``testing/``: Improvements or changes to testing
+-  ``release/``: Releases (see below)
+
+
+Pushing a New Branch
+--------------------
+Once you have implemented new changes and committed those, you need to push your
+branch. Pushing a branch means uploading your changes to the repository. These
+changes will only be seen by the branch you just created.
+
+.. code-block:: bash
+
+   git push -u origin <new branch name>
+
+Creating a Pull Request
+-----------------------
+Once you have tested your branch locally, create a PR and target your merge to
+``main``. This will automatically run CI testing and verify that your changes
+will work across all supported platforms. A detailed guideline of the process is
+collected under the `Creating a pull request
+<https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request>`_
+section in the GitHub documentation.
+
+For code verification, someone from the PyAnsys development team will review
+your code to verify that it meets the :ref:`Packaging Style`, :ref:`Coding
+Style`, :ref:`Documentation Style`.
+
+Once your code is approved, if you have write permission, you may merge the PR
+branch. If you don't have write permission, the reviewer or someone else with
+write permission will merge your PR and then delete your PR branch.
+
+.. admonition:: Always delete your PR branch after merging it into main branch
+
+   Deleting merged branches is a good practice which ensures that origin
+   branches do not get cluttered.
+
```

Comment:
This can be configured in the settings of the project to be done automatically. It should be enabled in all repos. We should include a warning saying this.